src/main/java/eu/yals/utils/ErrorUtils.java

```diff
@@ -0,0 +1,189 @@
+            YalsException yalsException = (YalsException) exception;
+
+            StringBuilder techMessage = new StringBuilder(yalsException.getMessage());
+            this.enrichTechMessageWithStatusAndPath(techMessage, args);
+
+            yalsErrorBuilder = YalsErrorBuilder.withTechMessage(techMessage.toString());
+
+            if (yalsException.hasMessageToUser()) {
+                yalsErrorBuilder.withMessageToUser(yalsException.getMessageToUser());
+            }
+
+            if (hasStatus) yalsErrorBuilder.withStatus(args.getStatus());
+
+        } else {
+            //general exception
+            StringBuilder techMessage = new StringBuilder(exception.getMessage());
+            this.enrichTechMessageWithStatusAndPath(techMessage, args);
+
+            yalsErrorBuilder = YalsErrorBuilder.withTechMessage(techMessage.toString());
+
+            //user message based on status
+            yalsErrorBuilder.withMessageToUser(UserMessageGenerator.getMessageByStatus(args.getStatus()));
+
+            if (hasStatus) yalsErrorBuilder.withStatus(args.getStatus());
+        }
+
+        yalsErrorBuilder.addRawException(exception);
+        return yalsErrorBuilder.build();
+    }
+
+    public void reportToBugsnag(YalsError yalsError) {
+        YalsException yalsException = new YalsException("Yals Error: " + yalsError.getId());
+        final String tabName = "Yals Error";
+        bugsnag.addCallback(report -> {
+            report.addToTab(tabName, "id", yalsError.getId());
+            report.addToTab(tabName, "Timestamp", yalsError.getTimeStamp());
+            report.addToTab(tabName, "Message to user", yalsError.getMessageToUser());
+            report.addToTab(tabName, "Tech Message", yalsError.getTechMessage());
+            report.addToTab(tabName, "HTTP Status", yalsError.getHttpStatus());
+            report.addToTab(tabName, "Raw Exception", yalsError.getRawException());
+        });
+        bugsnag.notify(yalsException);
+    }
+
+    public static Throwable findRootCause(Throwable throwable) {
+        if (throwable == null) return null;
+        Throwable rootCause = throwable;
+        while (rootCause.getCause() != null && rootCause.getCause() != rootCause) {
+            rootCause = rootCause.getCause();
+        }
+        return rootCause;
+    }
+
+    private void enrichTechMessageWithStatusAndPath(StringBuilder techMessage, Args args) {
+        boolean hasStatus = args.getStatus() != NO_STATUS;
+        boolean hasPath = StringUtils.isNotBlank(args.getPath());
+        if (hasStatus) {
+            techMessage.append(App.NEW_LINE).append("Status: ").append(args.getStatus());
+        }
+        if (hasPath) {
+            techMessage.append(App.NEW_LINE).append("Path: ").append(args.getPath());
+        }
+        techMessage.trimToSize();
+    }
+
+    @Data
+    public static class Args {
+        Throwable exception;
+        int status = NO_STATUS;
+        String path;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use explicit scoping instead of the default package private level](https://app.codacy.com/gh/yadevee/yals/pullRequest?prid=5392475)